code/create_spoken_forms.py

```diff
@@ -70,62 +71,90 @@ def create_spoken_forms(
         source: str,
         words_to_exclude: Optional[List[str]] = None,
         minimum_term_length: int = DEFAULT_MINIMUM_TERM_LENGTH,
+        generate_subsequences: bool = True,
     ) -> List[str]:
         """Create spoken forms for a given source"""
+
         if words_to_exclude is None:
             words_to_exclude = []
 
-        pieces = list(FULL_REGEX.finditer(source))
-        # print([piece.group(0) for piece in pieces])
-
-        term_sequence = " ".join(
-            [create_single_spoken_form(piece.group(0)) for piece in pieces]
-        ).split(" ")
-        # print(term_sequence)
-
-        terms = list(
-            {
-                term.lower().strip()
-                for term in (
-                    term_sequence
-                    + list(itertools.accumulate([f"{term} " for term in term_sequence]))
-                    + [source]
-                )
-            }
+        pieces_no_symbols = list(FULL_REGEX_NO_SYMBOLS.finditer(source))
+
+        pieces_with_symbols = list(FULL_REGEX.finditer(source))
+
+        spoken_form_with_symbols = " ".join(
+            [create_single_spoken_form(piece.group(0)) for piece in pieces_with_symbols]
+        ).lower()
+
+        spoken_form_without_symbols = " ".join(
+            [create_single_spoken_form(piece.group(0)) for piece in pieces_no_symbols]
+        ).lower()
+
+        # these two may be identical, so ensure the list is reduced
+        full_forms = list(
+            set(
+                [spoken_form_with_symbols.lower(), spoken_form_without_symbols.lower(),]
+            )
         )
 
+        # only generate the subsequences if requested
+        if generate_subsequences:
+            term_sequence = spoken_form_without_symbols.split(" ")
+            terms = list(
+                {
+                    term.lower().strip()
+                    for term in (
+                        term_sequence
+                        + list(
+                            itertools.accumulate([f"{term} " for term in term_sequence])
+                        )
+                    )
+                }
+            )
+
+            # add the full form if not present
+            if spoken_form_with_symbols not in terms:
+                terms.append(spoken_form_with_symbols)
+
+        else:
+            terms = full_forms
+
         terms = [
             term
             for term in terms
-            if term not in words_to_exclude and len(term) >= minimum_term_length
+            if (term not in words_to_exclude and len(term) >= minimum_term_length)
+            # always keep the full forms, even if < min term length?
```

Comment:
i think so, because otherwise there's no way to refer to this one right?

Reply:
Yeah, a simple test case with a folder name of "1" for instance with the file manager stuff